### PR TITLE
feat: Implement CLIENT LIST

### PR DESCRIPTION
**Pull request by siyopao**
Part of #174.

### CODE REVIEW

File: command.ts [hunk 1]
```diff
@@ -51,6 +51,13 @@ export interface BitfieldWithOverflowOpts extends BitfieldOpts {
 
 export type ClientCachingMode = "YES" | "NO";
 
+export interface ClientListOps {
+  type?: "NORMAL" | "MASTER" | "REPLICA" | "PUBSUB";
+  ids?: number[];
+}
+
+export type ClientPauseMode = "WRITE" | "ALL";
```
Comment (siyopao) [inline command.ts:59]: Sorry, moved this to keep it in alphabetical order.

File: redis.ts [hunk 2]
```diff
@@ -338,6 +339,19 @@ export class RedisImpl implements Redis {
     return this.execBulkReply("CLIENT", "INFO");
   }
 
+  clientList(opts?: ClientListOps) {
+    if (opts && opts.type && opts.ids) {
+      throw new Error("only one of `type` or `ids` may be specified");
```
Comment (uki00a) [inline redis.ts:344]:
```suggestion
      throw new Error("only one of `type` or `ids` can be specified");
```

**Review by uki00a — CHANGES_REQUESTED** [review]
Looks good! :+1: Just a minor nitpick:

File: redis.ts (hunk 2 above)
Comment (siyopao) [reply redis.ts]: Updated! Thanks!

**Review by uki00a — APPROVED** [review]
LGTM, thanks @siyopao! :+1:

**Pull request merged**